packages/block-editor/src/components/colors-gradients/dropdown.js

```diff
@@ -7,90 +7,175 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import {
-	__experimentalItemGroup as ItemGroup,
-	__experimentalItem as Item,
-	__experimentalHStack as HStack,
-	FlexItem,
+	Button,
 	ColorIndicator,
 	Dropdown,
+	FlexItem,
+	__experimentalHStack as HStack,
+	__experimentalItem as Item,
+	__experimentalItemGroup as ItemGroup,
+	__experimentalToolsPanelItem as ToolsPanelItem,
 } from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
 import ColorGradientControl from './control';
 
+const WithItemGroup = ( { isItemGroup, children } ) => {
+	if ( ! isItemGroup ) {
+		return children;
+	}
+
+	return (
+		<ItemGroup
+			isBordered
+			isSeparated
+			className="block-editor-panel-color-gradient-settings__item-group"
+		>
+			{ children }
+		</ItemGroup>
+	);
+};
+
+const WithToolsPanelItem = ( {
+	isItemGroup,
+	settings,
+	children,
+	...props
+} ) => {
+	if ( isItemGroup ) {
+		return children;
+	}
+
+	return (
+		<ToolsPanelItem
+			hasValue={ settings.hasValue }
+			label={ settings.label }
+			onDeselect={ settings.onDeselect }
+			isShownByDefault={ settings.isShownByDefault }
+			resetAllFilter={ settings.resetAllFilter }
+			{ ...props }
+			className="block-editor-tools-panel-color-gradient-settings__item"
+		>
+			{ children }
+		</ToolsPanelItem>
+	);
+};
+
+const renderToggleItem = ( { gradientValue, colorValue, label } ) => ( {
```

Comment:
You've already flagged it for a follow-up, but it looks like `renderToggleItem` and `renderToggleButton` could potentially be combined, but for the moment keeping them separate looks fine to me!

Reply:
Thanks, I should have some time today where I can clean this up while the PR awaits extra eyes.